src/plugins/console/public/services/autocomplete.ts

```diff
@@ -28,53 +54,174 @@ export class AutocompleteInfo {
+            title: i18n.translate('devTools.autocomplete.mappingResponseExceededTitle', {
+              defaultMessage: 'Maximum size of mappings response exceeded',
+            }),
+            text: i18n.translate('devTools.autocomplete.mappingResponseExceededText', {
+              defaultMessage:
+                'Some autocomplete suggestions may be missing. Disable autocomplete suggestions for fields in Settings to optimize performance.',
+            }),
+          });
+        }
+      }
+
+      this.pollTimeoutId = setTimeout(() => {
+        // This looks strange/inefficient, but it ensures correct behavior because we don't want to send
+        // a scheduled request if the user turns off polling.
+        if (settings.getPolling()) {
+          this.retrieve(settings, settings.getAutocomplete());
+        }
+      }, settings.getPollInterval());
+    });
+  }
+
+  private retrieveSettings<T>(
+    settingsKey: SettingsKey,
+    settingsToRetrieve: SettingsToRetrieve
+  ): Promise<T | {}> {
+    const settingKeyToPathMap = {
+      fields: '_mapping',
+      indices: '_aliases',
+      legacyTemplates: '_template',
+      indexTemplates: '_index_template',
+      componentTemplates: '_component_template',
+      dataStreams: '_data_stream',
+    };
+
+    // Fetch autocomplete info if setting is enabled and if user has made changes.
+    if (settingsToRetrieve[settingsKey]) {
+      return send({
+        http: this.http,
+        method: 'GET',
+        // pretty=false to compress the response and save bandwidth by avoiding whitespace
+        path: `${settingKeyToPathMap[settingsKey]}?pretty=false`,
+        asSystemRequest: true,
       });
+    } else {
+      // If the user doesn't want autocomplete suggestions, then clear any that exist
+      return Promise.resolve({});
+    }
+  }
+
+  private async retrieveMappings(settingsToRetrieve: SettingsToRetrieve) {
+    const mappings = await this.retrieveSettings<IndicesGetMappingResponse>(
+      ENTITIES.FIELDS,
+      settingsToRetrieve
+    );
+
+    if (mappings) {
+      this.mapping.loadMappings(mappings);
+    }
+  }
+
+  private async retrieveAliases(settingsToRetrieve: SettingsToRetrieve) {
+    const aliases = await this.retrieveSettings<IndicesGetAliasResponse>(
+      ENTITIES.INDICES,
+      settingsToRetrieve
+    );
+
+    if (aliases) {
+      const collaborator = this.mapping;
+      this.alias.loadAliases(aliases, collaborator);
+    }
+  }
+
+  private async retrieveDataStreams(settingsToRetrieve: SettingsToRetrieve) {
+    const dataStreams = await this.retrieveSettings<IndicesGetDataStreamResponse>(
+      ENTITIES.DATA_STREAMS,
+      settingsToRetrieve
+    );
+
+    if (dataStreams && 'data_streams' in dataStreams) {
+      this.dataStream.loadDataStreams(dataStreams);
+    }
+  }
+
+  private async retrieveTemplates(settingsToRetrieve: SettingsToRetrieve) {
+    const [legacyTemplates, indexTemplates, componentTemplates] = await Promise.all([
```

Comment:
can it be handled independently?

```
this.retrieveSettings<IndicesGetTemplateResponse>(
  ENTITIES.LEGACY_TEMPLATES, settingsToRetrieve
).then(() => {
  if (legacyTemplates) {
    this.legacyTemplate.loadTemplates(legacyTemplates);
  }
}) 

```
and so on

I can explain in case of an error of some promise, the current code will not process successful requests, you can probably use allSettled, but I think this needs to be fixed somehow